### PR TITLE
Ehn/cartopy

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

Adding cartopy as an optional dependency to pyDARN. 

Please note installing cartopy on various operating systems can be challenging. Please add any documentation that may help future users for your given system. I need to re-install it on OpenSuse so I will see if I need to go back and add information :p 

**issue:** #206 

## Approval

**Number of approvals:** 1-2 (tester and reviewer of documents)

## Test

**Note testers: please indicate what version of matplotlib you are using**

Testing is easy in that you just check if gives you warnings:
```
import pydarn
```
Output:
```
resetting environment variable IGRF_COEFFS in python script
resetting environment variable AACGM_v2_DAT_PREFIX in python script
non-default coefficient files may be specified by running aacgmv2.wrapper.set_coeff_path before any other functions

IMPORTANT!: If you are going to use Fan, Grid, and/or Convection Map plots, then make sure cartopy is installed on your machine. If you do not need to use cartopy for your plotting, ignore this message.
```

Testing new `Projs` enums
```
pydarn.Fan.plot_fan(fitacf_data, scan_index=2, projs=pydarn.Projs.COASTALINE,
                    colorbar_label='Velocity [m/s]',
                    line_color='red', radar_label=True, grid=True)
```
If cartopy is not installed (my case) should throw an exception:
```
resetting environment variable IGRF_COEFFS in python script
resetting environment variable AACGM_v2_DAT_PREFIX in python script
non-default coefficient files may be specified by running aacgmv2.wrapper.set_coeff_path before any other functions

IMPORTANT!: If you are going to use Fan, Grid, and/or Convection Map plots, then make sure cartopy is installed on your machine. If you do not need to use cartopy for your plotting, ignore this message.
An attempt to plot using cartopy was made without cartopy being installed. Please either change plot styles or install cartopy and dependencies.
Traceback (most recent call last):
  File "plot_fan.py", line 24, in <module>
    line_color='red', radar_label=True, grid=True)
  File "/home/marina/.local/lib/python3.6/site-packages/pydarn/plotting/fan.py", line 199, in plot_fan
    cls.plot_fov(dmap_data[0]['stid'], date, **kwargs)
  File "/home/marina/.local/lib/python3.6/site-packages/pydarn/plotting/fan.py", line 395, in plot_fov
    raise plot_exceptions.CartopyMissingError()
pydarn.exceptions.plot_exceptions.CartopyMissingError: An attempt to plot using cartopy was made without cartopy being installed. Please either change plot styles or install cartopy and dependencies.
```

I have not tested if cartopy is installed since we don't have a plotting method yet for it except in #203

### CODE REVIEW

File: pydarn/utils/projs.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#
+# This version of the GNU Lesser General Public License incorporates the terms
+# and conditions of version 3 of the GNU Lesser General Public License,
+# supplemented by the additional permissions listed below.
+#
+# Modifications:
+#
+
+"""
+coordinates.py is a group of methods that focus on coordinates systems used in
```
Comment (carleyjmartin) [inline pydarn/utils/projs.py:18]: update for projections.py

File: pydarn/utils/projs.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class Projs(enum.Enum):
+    """
+    Projection class is a list of enums for the various
+    projects supported by pyDARN.
+
+    enumerators:
+    """
+
+    POLAR = enum.auto()
+    COASTALINE = enum.auto
```
Comment (carleyjmartin) [inline pydarn/utils/projs.py:32]:
```suggestion
    COASTALINE = enum.auto()
```
I dont know fully what this does, but it just looked odd one having () and the other not.

**Review by carleyjmartin — COMMENTED** [review]
MacOS: 10.15.5
Python3.7
Matplotlib 3.4.2
Warning on import looks good, no issues installing or importing. 


I think there might be a missing ax kw somewhere in fov or fan, or maybe proj is making it go onto another axis I'm not sure, when plotting two fans giving the second one the first axes eg as `ax=ax1` it overplots the axes:
Screen Shot 2021-12-09 at 2 20 48 PM

```python
import matplotlib.pyplot as plt
import pydarn
import datetime as dt

#Read in fitACF file
file = "/Users/carley/Documents/data/20150305.0601.00.inv.fitacf"
SDarn_read = pydarn.SuperDARNRead(file)
fitacf_data = SDarn_read.read_fitacf()

time = dt.datetime(2015,3,5,6,1)
time2 = dt.datetime(2015,3,5,6,51)

plt.figure(figsize=(10,7))
#Make the fan plot and show
pydarn.Fan.plot_fan(fitacf_data, scan_index=time, lowlat=60, colorbar=True, groundscatter=True,
    zmin=-500, zmax=500, colorbar_label='Velocity', radar_label=True)

a = pydarn.Fan.plot_fan(fitacf_data, scan_index=time,colorbar_label='Velocity [m/s]',
        line_color='black', radar_label=True, groundscatter=1, grid = True, channel=0)
ax1 = plt.gca()
pydarn.Fan.plot_fan(fitacf_data, scan_index=time2,colorbar=False,
        line_color='blue', ax=ax1, radar_label=True, groundscatter=1, grid = True, channel=0)            

plt.show()
```

Back on develop branch the same code produces:
Screen Shot 2021-12-09 at 2 23 54 PM

Including/not including the new kw projs with `projs = pydarn.Projs.POLAR` which is default anyway works with no errors other than the axes above. 

Cartopy Missing Error works well when I try to do `projs = pydarn.Projs.COASTALINE`

I'm happy to write something up, but I basically followed this tutorial and it worked well: https://scitools.org.uk/cartopy/docs/latest/installing.html
and it would be very specifically for MacOS

Once cartopy is installed:
Then using 'projs = pydarn.Projs.COASTALINE' I get an error... but this is because this will be in another PR soon, yay!


For any other mac users and for reference, my installation process was:
Make sure you have homebrew installed - if not install it.
```
brew install proj geos
pip3 install --upgrade pyshp
pip3 uninstall shapely
pip3 install shapely --no-binary shapely
brew install pkg-config
export PKG_CONFIG_PATH=/usr/local/bin/pkgconfig
pip3 install cartopy
```
All this was done in a fresh venv `python3 -m virtualev venv-cartopy`

**Comment by mts299** [comment]
Gonna close this and will have a even better PR soon :)

**Pull request closed without merging**